### DISCUSSION TITLE
Collection ordering is chronologically inconsistent compared to history ordering

### DISCUSSION THREAD

**Discussion by hexylena** [galaxyproject/galaxy#18102] [General Discussion]
Collection | History
--- | ---
Older at top | Older at bottom
![Screenshot 2024-05-06 at 12-57-26 Galaxy](https://github.com/galaxyproject/galaxy/assets/458683/d95a5cc0-d865-4c96-83dd-1be49dfdd6dd) | ![Screenshot 2024-05-06 at 12-57-35 Galaxy](https://github.com/galaxyproject/galaxy/assets/458683/fefaf618-5de7-441c-af78-05c4a34b1e99)

yeah the numbering is correct but, it still feels inconsistent

**Comment by mvdbeek** [comment]
The history has no structure beyond when an item has been added, the collection does have an order that is important as that determines what elements get matched, and it does not correspond to age typically.

I think the best we could do is a button to switch the display order, but I'm not sure if this is making anything clearer ? 1 is the first element in the collection, and the first one to produce a new job when you match it against another collection, it would be kind of weird to reverse the order ?

**Comment by bernt-matthias** [comment]
I would like to add that from the tools point of view a collection (and it's elements) are created at one point in time (on discovery).

**Comment by hexylena** [comment]
> and it does not correspond to age typically.

this had not matched my experience but ok maybe I'm wrong there.

i just find it odd that they grow in different directions. if the history 'grew' downwards to match collections it would feel more consistent, it'd match the direction of written text, it'd match the ordering of the collection, etc. I guess we added to the top simply because it was easier than scrolling to bottom.

> I would like to add that from the tools point of view a collection (and it's elements) are created at one point in time (on discovery).

Only for output collections that are discovered, sure. But not mapping jobs.

**Comment by mvdbeek** [comment]
> Only for output collections that are discovered, sure. But not mapping jobs.

It's always true, also for mapping jobs. When the jobs are ready to be scheduled is an independent question (that's also true for history items).

**Comment by hexylena** [comment]
I find that y'all have a completely different mental model of collections from me as an end user of them, I'm guessing because you're working on the codebase of them more. For me i've always had it modelled as a folder, and I don't think I've given much thought to when the element was created, only when the datasets turn green because it's what matters to my usage of it.

I wish there was a better way that that could be communicated, because it'd probably decrease my issues every time I interact with them.

I'll move this issue to a discussion in case anyone else has the same disconnect in ordering and wishes to comment on it.

**Comment by bernt-matthias** [comment]
Would be good to have a good mental model documented somewhere. Mine is as follows

For tool generated collections there are two cases: 

- statically defined (I have never seen this except for pairs), here the tool developer defines the order
- discovered elements. Here there elements are defined at the end of the tool run. The ordering is defined in the tool (lexicographic, numeric, ...).

For collections generated in a map over the order is equal to the order of the input collection(s). All elements are created before any computing starts 

Collections are a bit like folders, but the important difference is that the content as well as the ordering is fixed.
